pyproject.toml

```diff
@@ -27,25 +17,43 @@ requires-python = ">=3.8"
 [project.scripts]
 canonical-sphinx-hello = "canonical_sphinx:hello"
 
+# Moving extended version to optional install, but first iteration will need to use previous full install
+# Once deployed on pypi, can change canonical-sphinx==0.1.0 to canonical-sphinx[full]
 [project.optional-dependencies]
+full = [
+    "linkify-it-py",
+    "canonical-sphinx-extensions",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "sphinx-notfound-page",
+    "sphinx-reredirects",
+    "sphinx-tabs",
+    "sphinxcontrib-jquery",
+    "sphinxext-opengraph",
+    "pyspelling",
+]
 dev = [
+    "canonical-sphinx==0.1.0",
     "build",
     "coverage[toml]==7.6.1",
     "pytest==8.3.2",
     "pytest-cov==5.0.0",
     "pytest-mock==3.14.0",
 ]
 lint = [
+    "canonical-sphinx==0.1.0",
     "black==24.8.0",
     "codespell[toml]==2.3.0",
     "ruff==0.5.7",
     "yamllint==1.35.1"
 ]
 types = [
+    "canonical-sphinx==0.1.0",
     "mypy[reports]==1.11.2",
     "pyright==1.1.378",
 ]
 docs = [
```

Comment:
The `[types]` optional dependencies are used by the `[testenv:lint-{mypy,pyright}]` step in tox.

The main package is the minimal install - though I think we could add an empty `[minimal]` install ref to potentially avoid confusion - but I don't think it would be that practical.

Reply:
You're right. Though as far as I can tell the `[docs]` dependency set was never actually needed for the RTD build. It could easily be bundled into `[full]`. We'd need to drop Python 3.8 support to bundle in the more recent versions of autobuild, but that's something we need to address anyway as 3.8 goes EOL.